tcms/xmlrpc/utils.py

```diff
@@ -34,10 +32,10 @@ def pre_check_product(values):
         if not product_str:
             raise ValueError('Got empty product name.')
         return Product.objects.get(name=product_str)
-    elif type(product_str) == int:  # to avoid breaking tests with bool(int)
+    # to avoid breaking tests with bool(int)
+    if type(product_str) == int:  # pylint: disable=unidiomatic-typecheck
```

Comment:
This has to stay, because `isinstance(True, int)` return `True`, while `type(True) == int` returns `False`, because returns `type(True) = <class 'bool'>`, which does is not equal to`int`

Reply:
This has to go, the only proper way of checking types is with the isinstance function. The test that will fail is not something which is likely to happen in real-life. This function is used to return a product object based on its name or ID. Nobody in their sane mind will send product=True via an API call and if they do, that's their problem.